### PR TITLE
KTOR-2150 Fix testUnsafeHeaders test

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt [hunk 1]
```diff
@@ -72,16 +72,22 @@ class HeadersTest : ClientLoader() {
...
-                "Header(s) ${HttpHeaders.UnsafeHeadersList} are controlled by the engine and cannot be set explicitly"
-            ) {
+            var message = ""
+            try {
                 client.get<HttpResponse>("$TEST_SERVER/headers") {
                     header(HttpHeaders.ContentLength, 0)
                     header(HttpHeaders.ContentType, ContentType.Application.Json)
                     header(HttpHeaders.TransferEncoding, "chunked")
                     header(HttpHeaders.Upgrade, "upgrade")
                 }
```
Comment (Stexxe) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt:82]: You can just `fail()` after `client.get()`

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt [hunk 2]
```diff
@@ -72,16 +72,22 @@ class HeadersTest : ClientLoader() {
...
+            } catch (cause: UnsafeHeaderException) {
+                message = cause.message ?: ""
```
Comment (Stexxe) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt:84]: Here you can insert an assertion

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt [hunk 3]
```diff
@@ -72,16 +72,22 @@ class HeadersTest : ClientLoader() {
...
             }
+
+            val expected =
+                "Header(s) ${HttpHeaders.UnsafeHeadersList} are controlled by the engine and cannot be set explicitly"
```
Comment (Stexxe) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt:88]: This line could be inlined to the `assertEquals` call

**Review by Stexxe — CHANGES_REQUESTED** [review]

**Comment by e5l** [comment]
Thanks. The requested changes are mostly about code style. From my perspective, having assertions at the bottom of the tests and extracting long strings in a separate variable are making tests better to read. Could you tell me if you have any strong objections to it?

**Review by Stexxe — APPROVED** [review]

**Pull request merged**